doc/docs.md

```diff
@@ -107,14 +107,14 @@ To do so, run the command `v up`.
     * [init functions](#init-functions)
 * [Type Declarations](#type-declarations)
     * [Interfaces](#interfaces)
+    * [Generics](#generics)
     * [Function Types](#function-types)
     * [Enums](#enums)
     * [Sum types](#sum-types)
     * [Type aliases](#type-aliases)
     * [Option/Result types & error handling](#optionresult-types-and-error-handling)
         * [Handling optionals/results](#handling-optionalsresults)
     * [Custom error types](#custom-error-types)
-    * [Generics](#generics)
```

Comment:
I disagree - imho interfaces and sumtypes should definitely be first, followed by the rest, and generics should be last - in my experience, unless you are writing specific kinds of libraries, not even applications, you can do so, without using any generics at all, while interfaces and sumtypes and the rest are very useful for most application and libraries.

Plus it helps, if you know what the basic types and their usages are, before writing generic functions over them.

Reply:
That is certainly one way to think about it.

On the other hand, with generics being considered one of the main features in a language that has them.
Although interfaces, generics, and thumb types have different specific uses, what they are trying to do can be classified in the same category.

One more thing.
I think Error Handling is more appropriate to be last in this list.
I don't think my view is absolutely correct, but I can't help but wonder about this area.

Reply:
By the way, here is the order I think is much better.

```markdown
* [Type Declarations](#type-declarations)
    * [Type aliases](#type-aliases)
    * [Enums](#enums)
    * [Function Types](#function-types)
    * [Interfaces](#interfaces)
    * [Sum types](#sum-types)
    * [Generics](#generics)
    * [Option/Result types & error handling](#optionresult-types-and-error-handling)
        * [Handling optionals/results](#handling-optionalsresults)
    * [Custom error types](#custom-error-types)
 ```

Reply:
I also think that "Type aliases" should be a little more complete.

Reply:
Generics should be last.

Reply:
Error handling is a basic feature of V, every V developer will encounter it sooner than later.

Reply:
> By the way, here is the order I think is much better.

That is a very good order, as long as Generics are last.

Reply:
Generics are much more abstract than all the rest. You can write many/most programs without using any generics.

Reply:
OK. 
I thought the Generic section was only listed last because of the order of publication, but if that is the clear intent, that is fine.

Reply:
> I thought the Generic section was only listed last because of the order of publication

That is a very reasonable assumption, but in this case, the last place is intentional.

Reply:
> On the other hand, with generics being considered one of the main features in a language that has it.

To me, the documentation (at least doc/docs.md), is *not an advertisement* for language snobs, that hop from one language to another, seeking an easy way to see what is different from the previous one, so that they can write blog posts about it, without much involvement. 

Instead, the documentation is a tool for *learning* the language, and should be prepared to help people, that want to learn and practice the language. Initially, for most, that happens by writing small apps in it, not libraries, and definitely not libraries that will benefit from V having generics.

Reply:
In that regard, my intention is not to advertise.
I refer to this document frequently, as my memory is not very good.
At that time, it is easier to find and understand if there is a good table of contents, or if related items are right next to each other, right?
That is the main motivation for these order edits.

Reply:
The same is true with regard to source code.
I find it easier to find and understand related code if it is in a logical order in close proximity to each other, so that I can see the whole picture.
Especially in situations where the IDE is not as rich in auxiliary tools, having the source code in a logical order and very well organized is very helpful to my work efficiency.

Reply:
It is very stressful for me, with my weak memory, when source code and documents are scattered all over the place without being organized in a reasonably well thought-out order and arrangement.

And I believe that such logically structured documentation and source code will help not only me but also others to work more comfortably and more efficiently.

Reply:
Yes, I agree - cohesive and complete documentation is very important.

Reply:
Generics however are abstractions over all other types, not just interfaces, sumtypes or any other particular type. They allow you to have abstract datatypes, like containers, and also functions/methods over them, specialized for each type that is later used, while still keeping most of the code the same.

You can look at them like related to interfaces, but that is not what they do - unlike interfaces that are mostly a runtime concept, generics do not exist at runtime at all. All of them get monomorphised at compile time by the V compiler into specialised instances.
 
They are more like templates of code, that is parametrized over types, not over values, where the types can be anything - primitive types like numbers, structs, sumtypes, interfaces, enums.

Reply:
This is a personal opinion unrelated to this document, but I believe that Interface, Generics, and SumTypes are all modern concepts designed to break the They all attempt to bypass the limitations of type specification and allow methods to be used.
They all attempt to bypass the limitations of type specification and allow methods to be used interchangeably.
From this perspective, I personally believe that learning all three as a set and understanding the different uses of each is helpful for effective programming.
Well, there are many different ways of thinking and perspectives. I think they are all correct in their own way.